pkg/services/ngalert/state/manager.go

```diff
@@ -245,7 +250,11 @@ func (st *Manager) DeleteStateByRuleUID(ctx context.Context, ruleKey ngModels.Al
 		s.SetNormal(reason, startsAt, now)
 		// Set Resolved property so the scheduler knows to send a postable alert
 		// to Alertmanager.
-		s.Resolved = oldState == eval.Alerting || oldState == eval.Error || oldState == eval.NoData
+		if oldState == eval.Alerting || oldState == eval.Error || oldState == eval.NoData {
+			s.ResolvedAt = now
+		} else {
+			s.ResolvedAt = time.Time{}
```

Comment:
I mostly decided on `time.Time{}` to both stay inline with `LastSentAt` which doesn't use a pointer and to keep closer to Prom ruler as it creates a much cleaner `NeedsSending` logic. For example, if both  `LastSentAt` and `ResolvedAt` were pointers, we'd need:

```go
	// We should send a notification if the state has been resolved since the last notification.
	if a.ResolvedAt != nil && (a.LastSentAt == nil || a.ResolvedAt.After(*a.LastSentAt)) {
		return true
	}
```
 instead of:

```go
	// We should send a notification if the state has been resolved since the last notification.
	if a.ResolvedAt.After(a.LastSentAt) {
		return true
	}
```
If I just use a pointer to `ResolvedAt` it's still fairly clean, however then it makes persisting them awkward as they'll need different treatment. WDYT?